### PR TITLE
[WIP] Mood Diary Models

### PR DESCRIPTION
**Pull request by severinkaderli**
* Add models for MoodDiary, Entry, Activity
* Add preliminary model for Patient
* Add ActivityType enum
* Add DAO for diaries
* Tests

Signed-off-by: Severin Kaderli <severin@kaderli.dev>

### CODE REVIEW

**Comment by severinkaderli** [comment]
I'm going to wait until #19 is merged before I continue this PR.

**Comment by martyschaer** [comment]
> I'm going to wait until #19 is merged before I continue this PR.

I'm on it, would you like to review?

**Comment by severinkaderli** [comment]
> > I'm going to wait until #19 is merged before I continue this PR.
> 
> I'm on it, would you like to review?

Yes, I will.

**Comment by severinkaderli** [comment]
@martyschaer 

I get the following error after I merged your changes from #19 into my branch and I try to run the tests.

```java
java.lang.ExceptionInInitializerError
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: org.hibernate.AnnotationException: @OneToOne or @ManyToOne on ch.bfh.bti7081.s2019.green.model.person.Patient.emergencyContacts references an unknown entity: java.util.List
```

**Comment by martyschaer** [comment]
> @martyschaer
> 
> I get the following error after I merged your changes from #19 into my branch and I try to run the tests.
> 
> ```java
> java.lang.ExceptionInInitializerError
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
> 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
> 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
> 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
> 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
> 	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
> 	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
> 	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
> 	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
> 	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
> Caused by: org.hibernate.AnnotationException: @OneToOne or @ManyToOne on ch.bfh.bti7081.s2019.green.model.person.Patient.emergencyContacts references an unknown entity: java.util.List
> ```

I've fixed several (in hindsight) obvious problems in #21.

**Comment by severinkaderli** [comment]
This PR is ready to merge as soon as @martyschaer has changed the database behaviour during tests, and the tests for this PR are running.

**Comment by martyschaer** [comment]
Can we disable "Too many instance variables" checks on codebeat for entities? Doesn't really make sense, does it?

**Comment by severinkaderli** [comment]
> Can we disable "Too many instance variables" checks on codebeat for entities? Doesn't really make sense, does it?

Yeah, I'm going to look into it.

File: src/main/java/ch/bfh/bti7081/s2019/green/model/diary/Activity.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import ch.bfh.bti7081.s2019.green.model.AbstractBaseEntity;
+import ch.bfh.bti7081.s2019.green.persistence.converters.LocalTimeConverter;
+import lombok.Data;
+import lombok.ToString;
+
+import javax.persistence.*;
+import java.time.LocalTime;
+
+@Data
+@ToString(exclude = {"entry"})
```
Comment (martyschaer) [inline src/main/java/ch/bfh/bti7081/s2019/green/model/diary/Activity.java:12]: This is cool, good to know

**Comment by severinkaderli** [comment]
Fixed it by adding a `.codebeatsettings` file in #23.

**Review by martyschaer — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**